cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/QueryResult.java

```diff
@@ -122,10 +122,16 @@ public DocumentRevision next() {
             if (!this.hasNext()) {
                 throw new NoSuchElementException();
             }
-            if (!documentBlock.hasNext()) {
+
+            DocumentRevision doc =  documentBlock.next();
```

Comment:
I wonder if there is a record for comment per line :)
I think that is helpful though.